### DISCUSSION TITLE
How to set the report only for one spec test using cypress-mochawesome-report

### DISCUSSION THREAD

**Discussion by adnanerlansyah403** [cypress-io/cypress#28440] [Questions and Help]
Is there any way that we can name the report of the test case specifically with the name of the spec test ?, for example, I have a spec test case for authentication & homepage and I only want to report the spec test authentication and not with the homepage. Is that possible?

**Comment by portyu9** [comment]
Hi @adnanerlansyah403 

The reporter reports the tests/specs that Cypress actually executes. If you only want an Authentication report, the cleanest solution is to run that spec separately.

For example:

```bash
npx cypress run \
  --spec "cypress/e2e/authentication.cy.js" \
  --reporter mochawesome \
  --reporter-options "reportDir=cypress/results/authentication,overwrite=false,html=true,json=true"
```

Then run the rest of the suite separately without that report, or with a different reporter/configuration.

For example:

```bash
npx cypress run --spec "cypress/e2e/homepage.cy.js"
```

If Authentication and Homepage tests currently live in the **same spec file**, split them into separate spec files first:

```text
cypress/e2e/authentication.cy.js
cypress/e2e/homepage.cy.js
```

Mochawesome is not intended to selectively omit arbitrary tests from a spec that Cypress already executed. Separating the specs/runs is more predictable and also lets you give each report its own output directory/name.

Please mark this answer as accepted if it helped, thank you.
